src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,353 @@
+            ctx.SaveNonEmptyString(_args.InputColumn);
+            ctx.SaveNonEmptyString(_args.OutputColumn);
+        }
+
+        public bool IsRowToRowMapper => true;
+
+        public IRowToRowMapper GetRowToRowMapper(ISchema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            return MakeDataTransform(new EmptyDataView(_host, inputSchema));
+        }
+
+        private sealed class Mapper : IRowMapper
+        {
+            private readonly IHost _host;
+            private readonly OnnxTransform _parent;
+
+            private readonly Type _outputItemRawType;
+            private readonly ColumnType _outputColType;
+            private readonly string _outputColName;
+
+            private readonly IdvToTensorAdapter _idvToTensorAdapter;
+
+            public Mapper(IHostEnvironment env, OnnxTransform parent, ISchema inputSchema)
+            {
+                Contracts.CheckValue(env, nameof(env));
+                _host = env.Register(nameof(Mapper));
+                _host.CheckValue(inputSchema, nameof(inputSchema));
+                _host.CheckValue(parent, nameof(parent));
+
+                _parent = parent;
+                var model = _parent.Model;
+                _idvToTensorAdapter = new IdvToTensorAdapter(inputSchema, parent._args.InputColumn,
+                                            model.ModelInfo.InputsInfo[0]);
+
+                // TODO: Remove assumption below
+                // Assume first output dimension is 1
+                var outputNodeInfo = model.ModelInfo.OutputsInfo[0];
+                int[] dims = outputNodeInfo.Shape.Skip(1).Select(x => (int)x).ToArray();
+                var outputItemType = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
+                _outputColType = new VectorType(outputItemType, dims);
+                _outputColName = _parent.Output;
+                _outputItemRawType = outputItemType.RawType;
+                _host.Assert(_outputItemRawType == _outputColType.ItemType.RawType);
+            }
+
+            public RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var info = new RowMapperColumnInfo[1];
+                info[0] = new RowMapperColumnInfo(_outputColName, _outputColType, null);
+                return info;
+            }
+
+            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
+            {
+                return col => activeOutput(0) && (_idvToTensorAdapter.IdvColumnIndex == col);
+            }
+
+            public void Save(ModelSaveContext ctx)
+            {
+                _parent.Save(ctx);
+            }
+
+            public Delegate[] CreateGetters(IRow input, Func<int, bool> activeOutput, out Action disposer)
+            {
+                disposer = null;
+                var getters = new Delegate[1];
+                using (var ch = _host.Start("CreateGetters"))
```

Comment:
> using (var ch = _host.Start("CreateGetters")) [](start = 15, length = 46)

what are you getting out of creating channel here?

Reply:
Mostly to keep  in sync with Tensorflow transform below. Yael may have better insight into this, perhaps for logging purposes.

[https://github.com/dotnet/machinelearning/blob/b83151659fea5cc2c228b6e92eb2717681a19764/src/Microsoft.ML.TensorFlow/TensorflowTransform.cs#L566](https://github.com/dotnet/machinelearning/blob/b83151659fea5cc2c228b6e92eb2717681a19764/src/Microsoft.ML.TensorFlow/TensorflowTransform.cs#L566)



---
In reply to: [220400265](https://github.com/dotnet/machinelearning/pull/942#discussion_r220400265) [](ancestors = 220400265)

Reply:
If there's a reason to remove it, we can remove as well -- I'm not sure what's the up/downsides for this.

---
In reply to: [220410822](https://github.com/dotnet/machinelearning/pull/942#discussion_r220410822) [](ancestors = 220410822,220400265)

Reply:
Reason: It doesn't do anything.

---
In reply to: [220410913](https://github.com/dotnet/machinelearning/pull/942#discussion_r220410913) [](ancestors = 220410913,220410822,220400265)

Reply:
Removed it.